src/GitVersionCore/BuildServers/AppVeyor.cs

```diff
@@ -23,11 +23,7 @@ public override string GenerateSetVersionMessage(VersionVariables variables)
 
             var data = string.Format("{{ \"version\": \"{0}.build.{1}\" }}", variables.FullSemVer, buildNumber);
             var bytes = Encoding.UTF8.GetBytes(data);
-            if (request.Headers == null)
```

Comment:
Is `request.Headers` never `null` since you deleted this?

Reply:
That is correct. It returns a pre-initialized collection, and even if you set it to null, it will fail when it tries to iterate the null value. See https://source.dot.net/#System.Net.Requests/System/Net/HttpWebRequest.cs,835